source/module_elecstate/potentials/pot_local.cpp

```diff
@@ -29,6 +29,12 @@ void PotLocal::cal_fixed_v(double *vl_pseudo // store the local pseudopotential
         }
     }
 
+    /// save the V_local at G=0
+    if(this->rho_basis_->npw > 0)
+    {
+        *vl_of_0_ = vg[0].real();
+    }
```

Comment:
is it possible to use such as  if (vg!=nullptr) instead of using npw?

Reply:
```
    std::complex<double> *vg = new std::complex<double>[this->rho_basis_->npw];
    ModuleBase::GlobalFunc::ZEROS(vg, this->rho_basis_->npw);
    for (int it = 0; it < this->ntype_; it++)
    {
        for (int ig = 0; ig < this->rho_basis_->npw; ig++)
        {
            vg[ig] += this->vloc_[0](it, this->rho_basis_->ig2igg[ig]) * this->sf_[0](it, ig);
        }
    }

//-------------------------------------------------
    /// save the V_local at G=0
    if(this->rho_basis_->npw > 0)
    {
        *vl_of_0_ = vg[0].real();
    }
//-------------------------------------------------

    const_cast<ModulePW::PW_Basis *>(this->rho_basis_)->recip2real(vg, vl_pseudo);
    delete[] vg;
```

Reply:
@dyzheng This is the context of this code.